### PR TITLE
UIP-2024 Add support for covariant prop / state fields

### PR DESCRIPTION
**Pull request by aaronlademann-wf**
> NOTE: This is not a clean diff.
>
> It will be clean once #75 merges

## Ultimate problem:
The `over_react` transformer did not support the `covariant` keyword recently added by the Dart SDK.

## How it was fixed:
The transformer was updated to detect the presence of the keyword, and place it within the setter parameter only.

over_react_covariant-prop-support

## Testing suggestions:
1. Verify that tests pass

## Potential areas of regression:
Transformed code

---

> __FYA:__ @Workiva/ui-platform-pp

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Raven
Number of Findings: 0

File: test/vm_tests/transformer/impl_generation_test.dart [hunk 1]
```diff
@@ -328,6 +328,19 @@ main() {
         ''');
       });
 
+test('covariant keyword', () {
```
Comment (jacehensley-wf) [inline test/vm_tests/transformer/impl_generation_test.dart]: Indentation is off here

**Comment by jacehensley-wf** [comment]
You'll also want to bump the travis.yaml dart SDK version

File: pubspec.yaml [hunk 2]
```diff
@@ -5,24 +5,25 @@ homepage: https://github.com/Workiva/over_react/
...
+  w_flux: "^2.7.1"
   platform_detect: "^1.1.1"
   quiver: ">=0.21.4 <0.25.0"
 dev_dependencies:
   browser_detect: "^1.0.4"
   matcher: ">=0.11.0 <0.13.0"
   coverage: "^0.7.2"
-  dart_dev: "^1.0.5"
+  dart_dev: "^1.7.6"
+  dart_style: ">=1.0.6 <2.0.0"
```
Comment (greglittlefield-wf) [inline pubspec.yaml]: Why was this added?

Comment (aaronlademann-wf) [inline pubspec.yaml]: mmm... must've added it when i was trying to get past the version solver issues w/ analyzer - which ended up being the `w_flux` package... will remove.

File: test/vm_tests/transformer/impl_generation_test.dart (hunk 1 above)
Comment (aaronlademann-wf) [inline test/vm_tests/transformer/impl_generation_test.dart]: facepalm

**Comment by aaronlademann-wf** [comment]
@jacehensley-wf I already updated the travis config file dart version.

**Comment by aaronlademann-wf** [comment]
@jacehensley-wf @greglittlefield-wf feedback addressed.

File: lib/src/transformer/impl_generation.dart [hunk 3]
```diff
@@ -296,7 +296,7 @@ class ImplGenerator {
             !member.isSynthetic &&
             member.isAbstract &&
             member.name.name == name &&
-            member.returnType?.name?.name == type
+            member.returnType == type
```
Comment (greglittlefield-wf) [inline lib/src/transformer/impl_generation.dart]: `member.returnType` is a `TypeAnnotation`, and `type` is a `String`... Not sure how this works.

For better future-proofing, can we change this to the following?
```dart
member.returnType?.toSource() == type
```

**Comment by jacehensley-wf** [comment]
+1

**Comment by greglittlefield-wf** [comment]
QA +10

- [X] Dev +1's
- [X] Dev/QA +10
- [X] Unit tests created/updated
- [X] All unit tests pass
- [X] Rosie ran/Rosie comment displays expected info
- [X] Dependency Scan Clean

Merging.
